src/Service/NetworksService.php

```diff
@@ -17,23 +17,40 @@ public function __construct(
         parent::__construct($repository, $mapper, $cache);
     }
 
-    public function findByUrlKeyWithDefaultService(string $urlKey): ?Network
+    public function findByUrlKeyWithDefaultService(string $urlKey, $ttl = CacheInterface::NORMAL): ?Network
     {
-        $dbEntity = $this->repository->findByUrlKeyWithDefaultService($urlKey);
-        return $this->mapSingleEntity($dbEntity);
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, $urlKey, $ttl);
+
+        return $this->cache->getOrSet(
+            $key,
+            $ttl,
+            function () use ($urlKey) {
+                $dbEntity = $this->repository->findByUrlKeyWithDefaultService($urlKey);
+                return $this->mapSingleEntity($dbEntity);
+            }
+        );
     }
 
     public function findPublishedNetworksByType(
         array $types,
         ?int $limit = self::DEFAULT_LIMIT,
-        int $page = self::DEFAULT_PAGE
+        int $page = self::DEFAULT_PAGE,
+        $ttl = CacheInterface::NORMAL
     ): array {
-        $dbEntities = $this->repository->findPublishedNetworksByType(
-            $types,
-            $limit,
-            $this->getOffset($limit, $page)
-        );
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, implode($types), $limit, $page, $ttl);
```

Comment:
It was my intention that each part of the key was split by a hyphen. If I imploded the types by hyphen too, it would not make the key as clear if we were to inspect it.